### PR TITLE
Update LED model card

### PR DESCRIPTION
**Pull request by dross20**
# What does this PR do?

This PR replaces the LED model card with a new model card matching the format introduced in https://github.com/huggingface/transformers/issues/36979.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?
@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/led.md [hunk 1]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
-  errors. This can be done by executing `model.gradient_checkpointing_enable()`. 
- Moreover, the `use_cache=False`
-  flag can be used to disable the caching mechanism to save memory.
-- LED is a model with absolute position embeddings so it's usually advised to pad the inputs on the right rather than
-  the left.
-
-This model was contributed by [patrickvonplaten](https://huggingface.co/patrickvonplaten).
+# LED
+
+[Longformer-Encoder-Decoder (LED)](https://huggingface.co/papers/2004.05150) is an encoder-decoder (sequence-to-sequence) transformer model for long-input summarization. It extends [Longformer](.longformer), an encoder-only model designed to handle long inputs, by adding a decoder layer for summarization tasks. Because of Longformer's linear self-attention mechanism, LED is more efficient than standard encoder-decoder models when summarizing long sequences.
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
[Longformer-Encoder-Decoder (LED)](https://huggingface.co/papers/2004.05150) is an encoder-decoder  transformer model for sequence-to-sequence tasks like summarization. It extends [Longformer](.longformer), an encoder-only model designed to handle long inputs, by adding a decoder layer. The decoder uses full self-attention on the encoded tokens and previously decoded locations. Because of Longformer's linear self-attention mechanism, LED is more efficient than standard encoder-decoder models when processing long sequences.
```

File: docs/source/en/model_doc/led.md [hunk 2]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+
+You can find all the original [LED] checkpoints under the [Ai2](https://huggingface.co/allenai/models?search=led) organization.
+
+> [!TIP]
+> This model was contributed by [patrickvonplaten](https://huggingface.co/patrickvonplaten).
+>
+> Click on the [LED] models in the right sidebar for more examples of how to apply [LED] to different language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
> Click on the LED models in the right sidebar for more examples of how to apply LED to different language tasks.
```

File: docs/source/en/model_doc/led.md [hunk 3]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+import torch
+from transformers import BitsAndBytesConfig, AutoModelForSeq2SeqLM, AutoTokenizer
+
+quantization_config = BitsAndBytesConfig(
+    load_in_4bit=True,
+    bnb_4bit_compute_dtype=torch.bfloat16,
+    bnb_4bit_quant_type="nf4"
+)
+model = AutoModelForSeq2SeqLM.from_pretrained(
+    "allenai/led-base-16384",
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
    "allenai/led-large-16384",
```

File: docs/source/en/model_doc/led.md [hunk 4]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+    torch_dtype=torch.bfloat16,
+    device_map="auto",
+    quantization_config=quantization_config
+)
+
+tokenizer = AutoTokenizer.from_pretrained(
+    "allenai/led-base-16384"
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
    "allenai/led-large-16384"
```

File: docs/source/en/model_doc/led.md [hunk 5]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+global_attention_mask[:, 0] = 1
+
+output = model.generate(**input_ids, global_attention_mask=global_attention_mask, cache_implementation="static")
+print(tokenizer.decode(output[0], skip_special_tokens=True))
+```
+
+## Notes
+
+- [`LEDForConditionalGeneration`] is an extension of [`BartForConditionalGeneration`] exchanging the traditional self-attention layer with Longformer's chunked self-attention layer. [`LEDTokenizer`] is an alias of [`BartTokenizer`].
+- LED pads the `input_ids` to be a multiple of `config.attention_window` if required. Therefore a small speed-up is gained when [`LEDTokenizer`] is used with the `pad_to_multiple_of` argument.
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
- LED pads the `input_ids` to be a multiple of `config.attention_window` if required. A small speedup is gained when [`LEDTokenizer`] is used with the `pad_to_multiple_of` argument.
```

File: docs/source/en/model_doc/led.md [hunk 6]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+- LED works best on long-range sequence-to-sequence tasks where the `input_ids` largely exceed a length of 1024 tokens.
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
- LED works best on long-range sequence-to-sequence tasks where the `input_ids` are significantly longer than 1024 tokens.
```

File: docs/source/en/model_doc/led.md [hunk 7]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+- LED makes use of global attention by means of the `global_attention_mask` (see [`LongformerModel`]). For summarization, it is advised to put global attention only on the first `<s>` token. For question answering, it is advised to put global attention on all tokens of the question.
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
- LED uses global attention by means of the `global_attention_mask` (see [`LongformerModel`]). For summarization, it is advised to put global attention only on the first `<s>` token. For question answering, it is advised to put global attention on all tokens of the question.
```

File: docs/source/en/model_doc/led.md [hunk 8]
```diff
@@ -14,62 +14,135 @@ rendered properly in your Markdown viewer.
...
+- To fine-tune LED on all 16384 parameters, gradient checkpointing can be enabled in case training leads to out-of-memory (OOM) errors. This can be done by executing `model.gradient_checkpointing_enable()`. Moreover, the `use_cache=False` flag can be used to disable the caching mechanism to save memory.
```
Comment (stevhliu) [inline docs/source/en/model_doc/led.md]:
```suggestion
- To fine-tune LED on all 16384 parameters, gradient checkpointing can be enabled in case training leads to out-of-memory (OOM) errors. Enable gradient checkpointing by adding `model.gradient_checkpointing_enable()` and setting `use_cache=False` to disable the caching mechanism to save memory.
```

**Review by stevhliu — APPROVED** [review]
Super nice, good job as always! 👏

**Comment by dross20** [comment]
Thank you for the feedback, @stevhliu! I applied your suggestions.
